### DISCUSSION TITLE
[Self-Hosting][Docker] Server and Data backup Question

### DISCUSSION THREAD

**Discussion by aqsede** [ente-io/ente#3009] [Q&A] [answered]
Hello,

First, I would like to say, ente is working great and very smooth. Uploading tens of thousand photo directly from Google Takeout seems to work wonders although I have no way idea how to verify if metadata is still detached. Anyway, just loving it.

My question is related to server and data backup incase my local server is completely dead.
I don't currently own or rent any S3 storage outside of the local bucket from the local configs.

What are things needed to be backed up aside from the ente_minio-data docker volume, credentials.yaml and compose file for smooth restoration?
Currently, I am just planning to do periodic rsync to a backup disk if that's okay. Through [ente's reliability](https://ente.io/reliability), I do understand we can already do multiple S3 instance to different drives natively (which I will figure out after some time) but the concern is still if any other part of the server is required to be backed up. Thanks for the time!

**Comment by mnvr** [comment, marked as answer]
You also need the database. See https://help.ente.io/self-hosting/faq/backup.

**Reply by aqsede** [reply]
thanks, I did check in FAQ, but missed this somehow... sorry for that.

**Reply by ahxxm** [reply]
Late to the question, I was examing breaking changes on `server/` and found these 3 values(not saying that they are breaking changes) `key.encryption, key.hash, jwt.secret`, I never set these values in credential yaml file, nor setup a public album, but the server works fine with a 2025-Feb HEAD, does [that means](https://github.com/ente-io/ente/blob/ee3f4451ed2010228a0607a2734bd4d8e526a49d/server/cmd/museum/main.go#L111) I'm using some default values?

**Reply by ahxxm** [reply]
(I do have separated backups of DB and B2 bucket, never tested a recovery though..)

**Reply by mnvr** [reply]
Yes, that'd mean you're using the defaults from [local.yaml](https://github.com/ente-io/ente/blob/1d2de2ab765d36d9a9b3cb9b3d895c3439786e6d/server/configurations/local.yaml#L199-L220)

**Reply by ahxxm** [reply]
oops!
